packages/backend/server/src/plugins/copilot/context/embedding.ts

```diff
@@ -0,0 +1,51 @@
+import OpenAI from 'openai';
+
+import { Embedding, EmbeddingClient } from './types';
+
+export class OpenAIEmbeddingClient extends EmbeddingClient {
+  constructor(private readonly client: OpenAI) {
+    super();
+  }
+
+  private cleanInput(input: string[]): string[] {
+    return input.map(i =>
+      i
+        // Remove unnecessary newlines and extra spaces
+        .replace(/\n+/g, ' ') // Merge multiple newlines into a single space
```

Comment:
slow

Reply:
remove this function or find a better way

Reply:
moved to rust

Reply:
the problem is not where it is put. deal with a string with regex again and again, how could it be fast

Reply:
they all could be done in a simple string iteration